### DISCUSSION TITLE
[Support]: frigate on proxmox lxc, trying to setup go2rtc.

### DISCUSSION THREAD

**Discussion by balajeek** [blakeblackshear/frigate#11627] [General Support]
### Describe the problem you are having

I am a long time user of frigate, awesome app that helps me a lot with the security of my property.

I use to run on docker on a ubuntu machine before, using 3 cameras on rtsp.

Recently moved to proxmox and setup a frigate lxc using the promox helper script, setup all went fine, i just moved my config and everything works the way it was.

Today I wanted to change the configuration, so instead of directly specifying my rtsp url under camera section like this:

```
  backyard:
    ffmpeg:
      inputs:
        - path: rtsp://cred@192.168.99.143/live0
          roles:
            - detect
            - rtmp
            - record
            - clips
```

into go2rtc method like below:
since i see go2trc is already bundled in the frigate, i updated my frigate config.yml to below
```
go2rtc:
  streams:
    backyard:
      - rtsp://cred@192.168.99.143/live0

cameras:
          
  backyard:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/backyard
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/backyard
          input_args: preset-rtsp-restream
          roles:
            - record
```

Now it does not work. here is the log from go2rtc:
```
[INFO] Preparing new go2rtc config...
[INFO] Starting go2rtc...
15:58:36.795 INF go2rtc version 1.9.0 linux/amd64
15:58:36.795 INF [api] listen addr=:1984
15:58:36.795 INF [rtsp] listen addr=:8554
15:58:36.795 INF [webrtc] listen addr=:8555
```

Here is the log from frigate: (I stripped down to keep it short)

```
[INFO] Preparing Frigate...
[INFO] Starting Frigate...
[2024-05-29 15:43:19] frigate.app                    INFO    : Starting Frigate (0.13.2-)
[2024-05-29 15:43:19] peewee_migrate.logs            INFO    : Starting migrations
[2024-05-29 15:43:19] peewee_migrate.logs            INFO    : There is nothing to migrate
[2024-05-29 15:43:19] frigate.app                    INFO    : Recording process started: 188779
[2024-05-29 15:43:19] frigate.app                    INFO    : go2rtc process pid: 139
[2024-05-29 15:43:19] detector.coral                 INFO    : Starting detection process: 188788
[2024-05-29 15:43:19] frigate.app                    INFO    : Output process started: 188794
[2024-05-29 15:43:19] frigate.app                    INFO    : Camera processor started for backyard: 188805
[2024-05-29 15:43:19] frigate.video                  ERROR   : backyard: Unable to read frames from ffmpeg process.
[2024-05-29 15:43:19] frigate.video                  ERROR   : backyard: ffmpeg process is not running. exiting capture thread...
[2024-05-29 15:43:39] ffmpeg.backyard.record         ERROR   : rtsp://127.0.0.1:8554/backyard: Server returned 404 Not Found
```

### Version

0.13.2

### Frigate config file

```yaml
mqtt:
  host: 192.168.99.100
  user: "user"
  password: "pass"

go2rtc:
  streams:
    backyard:
      - rtsp://cred@192.168.99.143/live0

cameras:
  backyard:
    ui:
      order: 3
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/backyard
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/backyard
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 1920
      height: 1080
      fps: 5
    objects:
      track:
        - person
        - animal
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: true
      retain:
        default: 3
        objects:
          person: 3
          animal: 3
    record:
      enabled: true
      retain:
        days: 3
      events:
        #post_capture: 10
        retain:
          default: 3
          objects:
            person: 3
            animal: 3
            vehicle: 3
    # motion:
    #  mask:
    #    - 760,671,1113,829,1492,1020,1920,1013,1920,1080,1562,1080,1443,1080,1358,1080,957,1080,0,1080,0,656
    #    - 0,0,0,269,1920,290,1920,213,1920,0,0,0
    # zones:
    #  zone_driveway:
    #    coordinates: 1376,446,1920,653,1920,986,1723,991,1530,992,1290,929,982,742,811,676,675,646,201,599,26,571,26,493



detectors:
  coral:
    type: edgetpu
    device: usb
  #cpu:
  #  type: cpu
```

### Relevant log output

```shell
provided above
```

### FFprobe output from your camera

```shell
na
```

### Operating system

Proxmox

### Install method

HassOS Addon

### Coral version

USB

### Network connection

Wired

### Camera make and model

many different

### Any other information that may be helpful

I beleive its not an issue with camera, becaz if i switch back to old config without using go2rtc it just works.
I am not sure if i have to do anything to setup go2rtc other than adding go2rtc: tag in the frigate config file.

please help.

**Comment by NickM-27** [comment]
what is your docker compose

**Reply by balajeek** [reply]
its not a docker compose, its directly installed on proxmox lxc.
I used this helper script from proxmox: https://helper-scripts.com/scripts?id=Frigate

**Reply by NickM-27** [reply]
this is installed in a way that we don't know about. There could be many problems including the way that script sets up networking, as well as how go2rtc is installed.

**Reply by balajeek** [reply]
when i login to the lxc console, i see go2rtc service is running and no errors.
frigate and camera all works fine until in try to change the config to use go2rtc.
So frigate ran fine with 3 cameras that i already setup

**Reply by NickM-27** [reply]
right, and we have no control or knowledge of how the network is configured for this installation. Maybe 127.0.0.1 (localhost) won't work, maybe something else is different.

**Comment by hawkeye217** [comment]
Many users have had success with proxmox, but we don't officially recommend it (nor do any of the maintainers use it) as a setup method. We're not quite sure what image you're running, as the tag version tag doesn't even show an ID.

**Reply by balajeek** [reply]
frigate and camera all works fine until today when i try to change the config to use go2rtc.
So frigate ran fine with 3 cameras that i already setup

**Reply by hawkeye217** [reply]
There are a fair number of discussion posts from users running proxmox, that may be your best way forward at this point.

**Comment by balajeek** [comment]
If i setup a go2rtc separately (either on my Home assistant addon or as another self hosted service)
can i stop the go2rtc that's bundled with frigate?

**Reply by NickM-27** [reply]
it won't be able to be used for live view and other features so for the most part no that is not an option

**Comment by lenne0815** [comment]
Its apparently a problem with file permissions inside the lxc:

https://github.com/tteck/Proxmox/discussions/3375

**Comment by C0DEbrained** [comment]
I was having this issue today on a fresh install. After adding `/usr/local/bin` to my PATH it seems to have fixed the problem. 

I'm guessing that some of the dependencies that the scripts rely on weren't being found so it was silently dying. Also, keep in mind that when using Frigate in this configuration, go2rtc doesn't seem to be restarted when Frigate is, so you'll need to restart that manually too to apply any config updates.
